src/audio/volume/volume.c

```diff
@@ -81,8 +81,17 @@ static enum task_state vol_work(void *data)
 	int again = 0;
 	int i;
 
-	/* inc/dec each volume if it's not at target */
-	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++) {
+	/* No need to ramp in idle state, jump volume to request. */
+	if (dev->state == COMP_STATE_READY) {
+		for (i = 0; i < PLATFORM_MAX_CHANNELS; i++)
+			cd->volume[i] = cd->tvolume[i];
+
+		vol_sync_host(cd, PLATFORM_MAX_CHANNELS);
+		return SOF_TASK_STATE_COMPLETED;
+	}
+
+	/* inc/dec each volume if it's not at target for active channels */
+	for (i = 0; i < cd->channels; i++) {
 		/* skip if target reached */
 		if (cd->volume[i] == cd->tvolume[i])
```

Comment:
They are now always in 0..N. We had some partial channel map implementation written by Liam and Keyon but it was not used by the driver so it was removed since it caused some bug a long time ago. So, this function need to be revisited such is re-implemented.

We have many volume instances and the volume ramp update is pretty frequent to avoid zipper noise so I prefer this function to be light.